jcl/src/java.base/share/classes/java/util/GPUAssistHolder.java

```diff
@@ -22,24 +22,30 @@
  *******************************************************************************/
 package java.util;
 
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+
 import com.ibm.gpu.spi.GPUAssist;
 
 final class GPUAssistHolder {
 
 	static final GPUAssist instance = gpuAssist();
 
 	private static GPUAssist gpuAssist() {
-		ServiceLoader<GPUAssist.Provider> loaded = ServiceLoader.load(GPUAssist.Provider.class);
-
-		for (GPUAssist.Provider provider : loaded) {
-			GPUAssist assist = provider.getGPUAssist();
-
-			if (assist != null) {
-				return assist;
+		GPUAssist gpuAssitRet = AccessController.doPrivileged(new PrivilegedAction<GPUAssist>() {
+			@Override
+			public GPUAssist run() {
+				ServiceLoader<GPUAssist.Provider> loaded = ServiceLoader.load(GPUAssist.Provider.class);
+		
+				for (GPUAssist.Provider provider : loaded) {
+					GPUAssist assist = provider.getGPUAssist();
+					if (assist != null) {
+						return assist;
+					}
+				}
+				return GPUAssist.NONE;
```

Comment:
Can you express this using a lambda (to reduce the verbosity)?  i.e.:
```
	PrivilegedAction<GPUAssist> action = () -> {
		ServiceLoader<GPUAssist.Provider> loaded = ServiceLoader.load(GPUAssist.Provider.class);

		for (GPUAssist.Provider provider : loaded) {
			GPUAssist assist = provider.getGPUAssist();

			if (assist != null) {
				return assist;
			}
		}

		return GPUAssist.NONE;
	};

	return AccessController.doPrivileged(action);
```

Reply:
Updated, pls have another look.